tests/RepositoryForms/Form/Processor/ContentFormProcessorTest.php

```diff
@@ -0,0 +1,271 @@
+
+        $field1 = $this->getField(self::FIELD_1_IDENTIFIER, self::LANGUAGE_CODE, 'field_1_value');
+        $field2 = $this->getField(self::FIELD_2_IDENTIFIER, self::LANGUAGE_CODE, 'field_2_value');
+
+        $data = new ContentUpdateData([
+            'contentDraft' => $this->contentDraft,
+            'fields' => [],
+            'fieldsData' => [
+                $field1->fieldDefIdentifier => new FieldData([
+                    'field' => $field1,
+                    'fieldDefinition' => $this->getFieldDefinition($field1->fieldDefIdentifier),
+                    'value' => 'new_field_1_value',
+                ]),
+                $field2->fieldDefIdentifier => new FieldData([
+                    'field' => $field2,
+                    'fieldDefinition' => $this->getFieldDefinition($field2->fieldDefIdentifier),
+                    'value' => 'new_field_2_value',
+                ]),
+            ],
+        ]);
+
+        $updateContentData = new ContentUpdateData([
+            'contentDraft' => $this->contentDraft,
+            'fields' => [
+                $this->getField(self::FIELD_1_IDENTIFIER, self::LANGUAGE_CODE, 'new_field_1_value'),
+                $this->getField(self::FIELD_1_IDENTIFIER, self::LANGUAGE_CODE, 'new_field_2_value'),
+            ],
+            'fieldsData' => [
+                $field1->fieldDefIdentifier => new FieldData([
+                    'field' => $field1,
+                    'fieldDefinition' => $this->getFieldDefinition($field1->fieldDefIdentifier),
+                    'value' => 'new_field_1_value',
+                ]),
+                $field2->fieldDefIdentifier => new FieldData([
+                    'field' => $field2,
+                    'fieldDefinition' => $this->getFieldDefinition($field2->fieldDefIdentifier),
+                    'value' => 'new_field_2_value',
+                ]),
+            ],
+        ]);
+
+        $event = new FormActionEvent($form, $data, 'foo');
+
+        $this->contentService
+            ->expects($this->once())
+            ->method('updateContent')
+            ->withConsecutive($this->versionInfo, $updateContentData)
+            ->willReturn($this->contentDraft);
+
+        $this->processor->processSaveDraft($event);
+    }
+
+    public function testProcessPublishWithOneUpdatedField(): void
+    {
+        $form = $this->getPublishForm();
+
+        $field1 = $this->getField(self::FIELD_1_IDENTIFIER, self::LANGUAGE_CODE, 'field_1_value');
+        $field2 = $this->getField(self::FIELD_2_IDENTIFIER, self::LANGUAGE_CODE, 'field_2_value');
+
+        $data = new ContentUpdateData([
+            'contentDraft' => $this->contentDraft,
+            'fields' => [],
+            'fieldsData' => [
+                $field1->fieldDefIdentifier => new FieldData([
+                    'field' => $field1,
+                    'fieldDefinition' => $this->getFieldDefinition($field1->fieldDefIdentifier),
+                    'value' => 'new_field_2_value',
```

Comment:
I think this should have different value. Other than that :+1: if this dosen't mess up with unchecking checkboxes for example.